pkg/apis/options/legacy_options.go

```diff
@@ -403,3 +414,52 @@ func getXAuthRequestAccessTokenHeader() Header {
 		},
 	}
 }
+
+type LegacyServer struct {
+	MetricsAddress string `flag:"metrics-address" cfg:"metrics_address"`
+	HTTPAddress    string `flag:"http-address" cfg:"http_address"`
+	HTTPSAddress   string `flag:"https-address" cfg:"https_address"`
+	TLSCertFile    string `flag:"tls-cert-file" cfg:"tls_cert_file"`
+	TLSKeyFile     string `flag:"tls-key-file" cfg:"tls_key_file"`
+}
+
+func legacyServerFlagset() *pflag.FlagSet {
+	flagSet := pflag.NewFlagSet("server", pflag.ExitOnError)
+
+	flagSet.String("metrics-address", "", "the address /metrics will be served on (e.g. \":9100\")")
```

Comment:
Do we want to add support for HTTPS metrics with the legacy flags?

I think both this PR & the initial metrics PR will go out together in the same release for the first time (v7.1.0 I imagine). So there's no backwards compatibility we need to preserve with the existing metrics flags.

TLS support seems handy if it isn't too burdensome.

Reply:
I guess distinct metrics tls file flags for solely the metrics endpoint won't help our config flag explosion problem 😨

Reply:
Totally happy to add them, should be relatively easy to add, what shall we call them :thinking: 
metrics-address, metrics-secure-address, metrics-tls-cert-file, metrics-tls-key-file? 

It's not too bad in terms of code management now these are in the legacy sets (as they are grouped nicely)

Reply:
Nice! Thanks for adding these!